src/Shared/E2ETesting/BrowserFixture.cs

```diff
@@ -77,12 +51,78 @@ public static bool IsHostAutomationSupported()
+                return (_browser, _logs);
             }
+
+            var opts = new ChromeOptions();
+
+            // Comment this out if you want to watch or interact with the browser (e.g., for debugging)
+            opts.AddArgument("--headless");
+
+            // Log errors
+            opts.SetLoggingPreference(LogType.Browser, LogLevel.All);
+
+            // On Windows/Linux, we don't need to set opts.BinaryLocation
+            // But for Travis Mac builds we do
+            var binaryLocation = Environment.GetEnvironmentVariable("TEST_CHROME_BINARY");
+            if (!string.IsNullOrEmpty(binaryLocation))
+            {
+                opts.BinaryLocation = binaryLocation;
+                output.WriteLine($"Set {nameof(ChromeOptions)}.{nameof(opts.BinaryLocation)} to {binaryLocation}");
+            }
+
+            var instance = await SeleniumStandaloneServer.GetInstanceAsync(output);
+
+            var attempt = 0;
+            var maxAttempts = 3;
+            do
+            {
+                try
+                {
+                    // The driver opens the browser window and tries to connect to it on the constructor.
+                    // Under heavy load, this can cause issues
+                    // To prevent this we let the client attempt several times to connect to the server, increasing
+                    // the max allowed timeout for a command on each attempt linearly.
+                    // This can also be caused if many tests are running concurrently, we might want to manage
+                    // chrome and chromedriver instances more aggresively if we have to.
+                    // Additionally, if we think the selenium server has become irresponsive, we could spin up
+                    // replace the current selenium server instance and let a new instance take over for the
+                    // remaining tests.
+                    var driver = new RemoteWebDriver(
+                        instance.Uri,
+                        opts.ToCapabilities(),
+                        TimeSpan.FromSeconds(60).Add(TimeSpan.FromSeconds(attempt * 60)));
+
+                    driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(1);
+                    var logs = new RemoteLogs(driver);
+
+                    _browser = driver;
+                    _logs = logs;
+
+                    return (_browser, _logs);
+                }
+                catch
+                {
+                    if (attempt >= maxAttempts)
+                    {
+                        throw new InvalidOperationException("Couldn't create a Selenium remote driver client. The server is irresponsive");
+                    }
+                }
+                attempt++;
+            } while (attempt < maxAttempts);
+
+            // We will never get here. Keeping the compiler happy.
+            throw new InvalidOperationException("Couldn't create a Selenium remote driver client. The server is irresponsive");
```

Comment:
nit: unresponsive would be the more standard word choice here, though irresponsive is accurate.